### DISCUSSION TITLE
Shape of MACE descriptor

### DISCUSSION THREAD

**Discussion by jungsdao** [ACEsuit/mace#160] [Q&A] [answered]
I have tried MACE descriptor and it gives different shape depending on the number of atoms in the system. 
In SOAP, regardless of the number of atom one can get fixed shape of descriptor which makes comparing across different systems viable. 
If I were to compare, systems with different atom number, how can this be possible with MACE descriptor??
Any comment would be appreciated! Thank you

**Comment by davkovacs** [comment]
MACE returns a descriptor for each atom in the system. If you want an overall descriptor the simplest thing is to average the per atom descriptors. This will give you something that has constant size.

**Comment by gabor1** [comment, marked as answer]
with SOAP, there are two ways to get a constant size descriptor that is independent of the number of atoms. Either you average the soap descriptors centered around each atom (this is the equivalent of what @davkovacs is suggesting above), or you can average the spherical harmonics expansion coefficients before they are squared to give rotational invariants. Both methods are available in the QUIP and the dscribe codes. This second method is not present out of the box in mace.
